File: frontend/src/api/stats.ts
import request from '@/utils/request'

export interface DashboardStats {
  total_tasks: number
  pending_count: number
  in_progress_count: number
  completed_count: number
  cancelled_count: number
  overdue_count: number
  today_completed: number
  this_week_completed: number
  by_priority: Record<string, number>
  recent_activity: Array<{
    task_title: string
    action: string
    time: string
  }>
}

export interface TrendData {
  labels: string[]
  values: number[]
}

export interface CategoryDistribution {
  name: string
  completed: number
  pending: number
  in_progress: number
}

export interface StatsCategoryDistribution {
  items: CategoryDistribution[]
}

// 仪表盘统计概览
export const getDashboardStats = () => {
  return request.get('/stats/dashboard') as Promise<DashboardStats>
}

// 趋势统计
export const getTrendStats = (period: string = 'week', field: string = 'completed') => {
  return request.get('/stats/trend', { params: { period, field } }) as Promise<TrendData>
}

// 分类分布
export const getCategoryDistribution = () => {
  return request.get('/stats/category-distribution') as Promise<StatsCategoryDistribution>
}

File: backend/app/schemas/stats.py
"""统计模块 Pydantic schemas"""

from pydantic import BaseModel


class DashboardStats(BaseModel):
    total_tasks: int
    pending_count: int
    in_progress_count: int
    completed_count: int
    cancelled_count: int
    overdue_count: int
    today_completed: int
    this_week_completed: int
    by_priority: dict[str, int]
    recent_activity: list[dict]


class TrendData(BaseModel):
    labels: list[str]
    values: list[int]


class CategoryDistribution(BaseModel):
    name: str
    completed: int
    pending: int
    in_progress: int


class StatsCategoryDistribution(BaseModel):
    items: list[CategoryDistribution]
